sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SyntaxTrackerBase.cs

```diff
@@ -50,13 +54,7 @@ protected bool IsEnabled(AnalyzerOptions options)
                 return false;
             }
 
-            return supportedDiagnostics.Any(d => analysisConfiguration.IsEnabled(d.Id));
+            return analysisConfiguration.IsEnabled(Rule.Id);
```

Comment:
Not changed in this PR, but there are two checks for _analysisConfiguration.EnabledRules == null_ earlier in this method (lines 47 and 52).